src/ChunkedStreamDecoder.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace React\HttpClient;
+
+use Evenement\EventEmitterTrait;
+use React\Stream\DuplexStreamInterface;
+use React\Stream\Util;
+
+class ChunkedStreamDecoder
+{
+    const CRLF = "\r\n";
+
+    use EventEmitterTrait;
+
+    /**
+     * @var string
+     */
+    protected $buffer = '';
+
+    /**
+     * @var int
+     */
+    protected $remainingLength = 0;
+
+    /**
+     * @var bool
+     */
+    protected $nextChunkIsLength = true;
+
+    /**
+     * @var DuplexStreamInterface
+     */
+    protected $stream;
+
+    /**
+     * @param DuplexStreamInterface $stream
+     */
+    public function __construct(DuplexStreamInterface $stream)
+    {
+        $this->stream = $stream;
+        $this->stream->on('data', array($this, 'handleData'));
+        Util::forwardEvents($this->stream, $this, [
+            'error',
+            'end',
+        ]);
+    }
+
+    public function handleData($data)
+    {
+        $this->buffer .= $data;
+
+        do {
+            $this->iterateBuffer();
+        } while (strlen($this->buffer) > 0 && strpos($this->buffer, static::CRLF) !== false);
+    }
+
+    protected function iterateBuffer()
+    {
+        if ($this->nextChunkIsLength) {
+            $this->nextChunkIsLength = false;
+            $this->remainingLength = hexdec(substr($this->buffer, 0, strpos($this->buffer, static::CRLF)));
```

Comment:
If i read it correctly, node ignores it: https://github.com/nodejs/http-parser/blob/master/http_parser.c#L1989
I'd vote for ignoring it for now. If requested it could be emitted as a third parameter of the `data` event.